appserver/connectors/connectors-internal-api/src/main/java/com/sun/appserv/connectors/internal/api/ConnectorsUtil.java

```diff
@@ -111,7 +111,7 @@ public static boolean belongsToSystemRA(String raName) {
 
     public static boolean getPingDuringPoolCreation(PoolInfo poolInfo, Resources allResources) {
         ResourcePool pool = getConnectionPoolConfig(poolInfo, allResources);
-        return Boolean.parseBoolean(pool.getPing());
+        return pool != null && Boolean.parseBoolean(pool.getPing());
```

Comment:
Theoretically `getConnectionPoolConfig` can return null which I thought defaults to `false`. Didn't wanted to make this more complicated with logging or exceptions. So NPE becomes `false`.

Reply:
`ResourcePool#getPing` has a default value, so I believe it cannot return null.

Reply:
I think you are - also `parseBoolean` could handle a null value.